src/_pytest/config/argparsing.py

```diff
@@ -232,6 +216,29 @@ def addini(
         self._inidict[name] = (help, type, default)
         self._ininames.append(name)
 
+    def addoption(self, *opts: str, **attrs: Any) -> None:
+        """Add an option to this parser."""
+        # Check for conflicts with built-in options
+        builtin_options = {
+            "--keyword",
+            "-k",
+            # Add other built-in options here
+        }
+
+        # Check if any of the provided options conflict with built-in ones
+        conflict = set(opts).intersection(builtin_options)
```

Comment:
instead of tracking builtin ones, we really should track registrations and warn/error early

currently what we do is defer errors to when we compose argparse